src/com/boxboat/jenkins/pipeline/promote/BoxPromote.groovy

```diff
@@ -135,8 +135,13 @@ class BoxPromote extends BoxBase<PromoteConfig> implements Serializable {
                 fullSemVer = "${config.gitTagPrefix}${fullSemVer}"
             }
 
-            if (gitRepo.getTagReferenceHash(fullSemVer) == gitCommitToTag ||
-                    gitRepo.getTagReferenceHash(fullSemVer) == gitRepo.getTagReferenceHash(fullGitTagToTag)) {
+            // Ensure non-null values so check isn't null == null
```

Comment:
I think a null check on `gitRepo.getTagReferenceHash(fullSemVer)` only would satisfy ensuring `null != null`

since `gitRepo.getTagReferenceHash(fullSemVer)` is a "computationally expensive" call (executes shell code on agent), can you save it off to a variable?  this should have been done before but now is a good opportunity to fix.  e.g.

```
            def semVerReferenceHash = gitRepo.getTagReferenceHash(fullSemVer)

            if (semVerReferenceHash != null && (
                    semVerReferenceHash == gitCommitToTag ||
                    semVerReferenceHash == gitRepo.getTagReferenceHash(fullGitTagToTag))) {
```

Reply:
Updated